src/ray/gcs/gcs_server/gcs_placement_group_manager.cc

```diff
@@ -509,7 +511,7 @@ void GcsPlacementGroupManager::CleanPlacementGroupIfNeededWhenActorDead(
       continue;
     }
     placement_group->MarkCreatorActorDead();
-    if (placement_group->IsPlacementGroupRemovable()) {
+    if (!placement_group->IsDetached() && placement_group->IsPlacementGroupRemovable()) {
```

Comment:
Why don't we modify the function name `IsPlacementGroupRemovable` to `IsPlacementGroupLifetimeDone` and move `placement_group->IsDetached()` into that method?

Reply:
That is a good idea